src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -138,14 +138,12 @@ void BaseCouplingScheme::sendData(const m2n::PtrM2N &m2n, const DataMap &sendDat
       }
     } else {
       if (data->hasGradient()) {
-        data->sample() = stamples.back().sample;
         // Data is only received on ranks with size>0, which is checked in the derived class implementation
-        m2n->send(data->values(), data->getMeshID(), data->getDimensions());
-        m2n->send(data->gradients(), data->getMeshID(), data->getDimensions() * data->meshDimensions());
+        m2n->send(data->stamples().back().sample.values, data->getMeshID(), data->getDimensions());
+        m2n->send(data->stamples().back().sample.gradients, data->getMeshID(), data->getDimensions() * data->meshDimensions());
```

Comment:
Do we know the timestamp of this sample?, then we could avoid `back()`

Reply:
`getTime()` should work. It's behavior is a bit complicated since it depends on the context where `sendData/receiveData` is called. But I think it's still a step into the right direction.

I will also add some assertions to make the context-sensitivity more explicit.